electron-app/src/background.ts

```diff
@@ -81,27 +82,112 @@ function createWindow() {
     win.loadURL('app://./index.html');
   }
 
-  // Check if we are on a MAC
-  if (process.platform === 'darwin') {
-    // Create our menu entries so that we can use MAC shortcuts
-    const template: MenuItemConstructorOptions[] = [
-      {
-        label: 'Edit',
-        submenu: [
-          { role: 'undo' },
-          { role: 'redo' },
-          { type: 'separator' },
-          { role: 'cut' },
-          { role: 'copy' },
-          { role: 'paste' },
-          { role: 'pasteAndMatchStyle' },
-          { role: 'delete' },
-          { role: 'selectAll' }
+  const defaultMenuTemplate = [
+    // On a mac we want to show a "Rotki" menu item in the
+    ...(isMac
+      ? [
+          {
+            label: app.name,
+            submenu: [
+              { role: 'hide' },
+              { role: 'hideOthers' },
+              { role: 'unhide' },
+              { type: 'separator' },
+              { role: 'quit' }
+            ]
+          }
         ]
-      }
-    ];
-    Menu.setApplicationMenu(Menu.buildFromTemplate(template));
-  }
+      : []),
+    {
+      label: 'File',
+      submenu: [isMac ? { role: 'close' } : { role: 'quit' }]
+    },
+
+    {
+      label: '&Edit',
+      submenu: [
+        { role: 'undo' },
+        { role: 'redo' },
+        { type: 'separator' },
+        { role: 'cut' },
+        { role: 'copy' },
+        { role: 'paste' },
+        // Macs have special copy/paste and speech functionality
+        ...(isMac
+          ? [
+              { role: 'pasteAndMatchStyle' },
+              { role: 'delete' },
+              { role: 'selectAll' },
+              { type: 'separator' },
+              {
+                label: 'Speech',
+                submenu: [{ role: 'startspeaking' }, { role: 'stopspeaking' }]
+              }
+            ]
+          : [{ role: 'delete' }, { type: 'separator' }, { role: 'selectAll' }])
+      ]
+    },
+    {
+      label: '&View',
+      submenu: [
+        { role: 'reload' },
+        { role: 'forceReload' },
+        { role: 'toggleDevTools' },
+        { type: 'separator' },
+        { role: 'resetZoom' },
+        { role: 'zoomIn' },
+        { role: 'zoomOut' },
+        { type: 'separator' },
+        { role: 'togglefullscreen' }
+      ]
+    },
+    {
+      label: '&Help',
+      submenu: [
```

Comment:
For get premium I would guess at least for now it could have its own submenu? Also ... I am exploring the possibility of getting sponsors for the app so a separate sponsors submenu, which would open a popup showing the logos of sponsors would be cool. 

I am not sure when do the submenus become too many :)

Reply:
Another possibility: on macs we have a "Rotki" menu. In our case File actually doesn't do much (just access to 'Exit'). We could have a "Rotki" menu in windows/linux as well and merge the two and put "Get Premium" and "Sponsors" in there.

Reply:
Perhaps. I would have to see it. The Get Premium should be really easily visible though. That's why not sure if it is good to have it in a different submenu. I guess paying sponsors would expect the same.